deps/v8/src/sampler.cc

```diff
@@ -342,6 +342,14 @@ void SignalHandler::HandleProfilerSignal(int signal, siginfo_t* info,
     // We require a fully initialized and entered isolate.
     return;
   }
+
+  // ISSUE-14576: To avoid deadlock, return if there is a thread lock
+  if (Isolate::GetProcessWideMutexPointer()->TryLock()) {
```

Comment:
@dmelikyan Right, I understand the rationale behind your fix now, thank you for the clarification. 

What is already locking the process wide mutex in the profiler thread when `SignalHandler::HandleProfilerSignal` is called? Possibly related question: is `SIGPROF` blocked while running `SignalHandler::HandleProfilerSignal`? Otherwise I imagine there would still be a race condition.